### PR TITLE
#9250 Refactor ModActionItem remove unnecessary props

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Part of #9250 (preliminary refactor)
- Removes the majority of props on ModActionItem that are better suited to instead be defined within the scope of the component itself
- No functional changes to the product

### CODE REVIEW

File: src/pageEditor/modListingPanel/ModActionMenu.tsx [hunk 1]
```diff
@@ -28,79 +28,87 @@ import styles from "./ActionMenu.module.scss";
...
   return (
     <div className={styles.root}>
-      {onSave != null && (
-        <SaveButton
-          ariaLabel={labelRoot ? `${labelRoot} - Save` : undefined}
-          onClick={onSave}
-          disabled={!isDirty}
-        />
-      )}
+      {/* TODO: did we really want to always show SaveButton? That is the current behavior as of 2.1.5-beta.1 */}
```
Comment (mnholtz) [inline src/pageEditor/modListingPanel/ModActionMenu.tsx:104]: Do we? I noticed that the disabled save button might look a little awkward, but maybe that's just me

Screenshot 2024-10-08 at 6 28 11 PM

Comment (mnholtz) [reply src/pageEditor/modListingPanel/ModActionMenu.tsx:104]: I would actually sooner expect the ellipsis menu to always show, rather than the save button

Comment (twschiller) [reply src/pageEditor/modListingPanel/ModActionMenu.tsx:104]: @mnholtz we had decided to run with this UI/UX for now. If you have questions, please raise in `#ui-ux-design channel`

**Review by twschiller — APPROVED** [review]
Thanks for the PR. We could also apply similar refactoring to ModComponentActionMenu in the future

Before merging let's remove comment https://github.com/pixiebrix/pixiebrix-extension/pull/9254/files#r1792674185 and move that discussion to the appropriate channel

**Pull request merged**